src/main/java/ai/elimu/rest/v2/crowdsource/WordContributionRestController.java

```diff
@@ -103,45 +95,50 @@ public String handleWordContributionRequest(
-        logger.info("wordGson: " + wordGson);
+
+        // Convert the request body to a WordContributionEventGson
+        WordContributionEventGson wordContributionEventGson =
+                new Gson().fromJson(requestBody, WordContributionEventGson.class);
+        logger.info("wordGson: " + wordContributionEventGson);
+
+        // Extract the WordGson from the WordContributionEventGson
+        WordGson wordGson = wordContributionEventGson.getWord();
 
         // Check if the word is already existing.
         Word existingWord = wordDao.readByText(wordGson.getText());
         if (existingWord != null) {
             jsonObject.put("result", "error");
             jsonObject.put("errorMessage", "NonUnique");
-            response.setStatus(HttpStatus.BAD_REQUEST.value());
+            response.setStatus(HttpStatus.CONFLICT.value());
 
             String jsonResponse = jsonObject.toString();
             logger.info("jsonResponse: " + jsonResponse);
             return jsonResponse;
         }
 
         try {
-            //Convert the JSON String to WordGson Object.
-            Word newWord = new Word();
-            newWord.setWordType(wordGson.getWordType());
-            newWord.setText(wordGson.getText());
-            List<LetterToAllophoneMappingGson> wordGsonLetterToAllophoneMappingsGson = wordGson.getLetterToAllophoneMappings();
+            // Convert the WordGson to Word POJO.
+            Word word = new Word();
+            word.setWordType(wordGson.getWordType());
+            word.setText(wordGson.getText());
+            List<LetterToAllophoneMappingGson> letterToAllophoneMappingsGsons = wordGson.getLetterToAllophoneMappings();
             List<LetterToAllophoneMapping> letterToAllophoneMappings = new ArrayList<>();
-            for (LetterToAllophoneMappingGson letterToAllophoneMappingGson : wordGsonLetterToAllophoneMappingsGson) {
-                LetterToAllophoneMapping letterToAllophoneMapping = letterToAllophoneMappingDao.read(letterToAllophoneMappingGson.getId());
+            for (LetterToAllophoneMappingGson letterToAllophoneMappingGson : letterToAllophoneMappingsGsons) {
+                LetterToAllophoneMapping letterToAllophoneMapping =
+                        letterToAllophoneMappingDao.read(letterToAllophoneMappingGson.getId());
                 letterToAllophoneMappings.add(letterToAllophoneMapping);
             }
-            newWord.setLetterToAllophoneMappings(letterToAllophoneMappings);
-            wordDao.create(newWord);
-
-            //Convert Request Body(String) to JSON Object.
-            JSONObject requestJson = new JSONObject(requestBody);
+            word.setLetterToAllophoneMappings(letterToAllophoneMappings);
+            wordDao.create(word);
 
             WordContributionEvent wordContributionEvent = new WordContributionEvent();
             wordContributionEvent.setContributor(contributor);
             wordContributionEvent.setTime(Calendar.getInstance());
```

Comment:
```suggestion
            wordContributionEvent.setTime(wordContributionEventGson.getTime());
```

Strictly speaking we should use `wordContributionEvent.setTime(wordContributionEventGson.getTime());` instead of `wordContributionEvent.setTime(Calendar.getInstance());` because this timestamp refers to the event in the Android app itself and not on the server-side, and theoretically there could be a little delay between the creation event and the server processing the request.